### PR TITLE
[Core] adding missing variables synchronization in Communicator

### PR DESCRIPTION
**Pull request by philbucher**
adding synchronization for the variables that were added in #2257
needed for #3656

### CODE REVIEW

File: kratos/includes/communicator.h [hunk 1]
```diff
@@ -623,6 +638,21 @@ class Communicator
         return true;
     }
 
+    virtual bool SynchronizeNonHistoricalVariable(Variable<array_1d<double, 4 > > const& ThisVariable)
```
Comment (adityaghantasala) [inline kratos/includes/communicator.h]:
```suggestion
    virtual bool SynchronizeNonHistoricalVariable(Variable<array_1d<double, 4 > > const& rThisVariable)
```

File: kratos/includes/communicator.h [hunk 2]
```diff
@@ -623,6 +638,21 @@ class Communicator
...
+    {
+        return true;
+    }
+
+    virtual bool SynchronizeNonHistoricalVariable(Variable<array_1d<double, 6 > > const& ThisVariable)
```
Comment (adityaghantasala) [inline kratos/includes/communicator.h]:
```suggestion
    virtual bool SynchronizeNonHistoricalVariable(Variable<array_1d<double, 6 > > const& rThisVariable)
```

File: kratos/includes/communicator.h [hunk 3]
```diff
@@ -623,6 +638,21 @@ class Communicator
...
+    {
+        return true;
+    }
+
+    virtual bool SynchronizeNonHistoricalVariable(Variable<array_1d<double, 9 > > const& ThisVariable)
```
Comment (adityaghantasala) [inline kratos/includes/communicator.h]:
```suggestion
    virtual bool SynchronizeNonHistoricalVariable(Variable<array_1d<double, 9 > > const& rThisVariable)
```

File: kratos/mpi/includes/mpi_communicator.h [hunk 4]
```diff
@@ -547,6 +547,24 @@ class MPICommunicator : public Communicator
         return true;
     }
 
+    bool SynchronizeVariable(Variable<array_1d<double, 4 > > const& ThisVariable) override
```
Comment (adityaghantasala) [inline kratos/mpi/includes/mpi_communicator.h]:
```suggestion
    bool SynchronizeVariable(Variable<array_1d<double, 4 > > const& rThisVariable) override
```

File: kratos/mpi/includes/mpi_communicator.h [hunk 5]
```diff
@@ -547,6 +547,24 @@ class MPICommunicator : public Communicator
...
+    {
+        SynchronizeVariable<array_1d<double, 4 >,double >(ThisVariable);
```
Comment (adityaghantasala) [inline kratos/mpi/includes/mpi_communicator.h]:
```suggestion
        SynchronizeVariable<array_1d<double, 4 >,double >(rThisVariable);
```

File: kratos/mpi/includes/mpi_communicator.h [hunk 6]
```diff
@@ -547,6 +547,24 @@ class MPICommunicator : public Communicator
...
+        return true;
+    }
+
+    bool SynchronizeVariable(Variable<array_1d<double, 6 > > const& ThisVariable) override
```
Comment (adityaghantasala) [inline kratos/mpi/includes/mpi_communicator.h]:
```suggestion
    bool SynchronizeVariable(Variable<array_1d<double, 6 > > const& rThisVariable) override
```

File: kratos/mpi/includes/mpi_communicator.h [hunk 7]
```diff
@@ -547,6 +547,24 @@ class MPICommunicator : public Communicator
...
+    {
+        SynchronizeVariable<array_1d<double, 6 >,double >(ThisVariable);
```
Comment (adityaghantasala) [inline kratos/mpi/includes/mpi_communicator.h]:
```suggestion
        SynchronizeVariable<array_1d<double, 6 >,double >(rThisVariable);
```

File: kratos/mpi/includes/mpi_communicator.h [hunk 8]
```diff
@@ -547,6 +547,24 @@ class MPICommunicator : public Communicator
...
+        return true;
+    }
+
+    bool SynchronizeVariable(Variable<array_1d<double, 9 > > const& ThisVariable) override
```
Comment (adityaghantasala) [inline kratos/mpi/includes/mpi_communicator.h]:
```suggestion
    bool SynchronizeVariable(Variable<array_1d<double, 9 > > const& rThisVariable) override
```

File: kratos/mpi/includes/mpi_communicator.h [hunk 9]
```diff
@@ -547,6 +547,24 @@ class MPICommunicator : public Communicator
...
+    {
+        SynchronizeVariable<array_1d<double, 9 >,double >(ThisVariable);
```
Comment (adityaghantasala) [inline kratos/mpi/includes/mpi_communicator.h]:
```suggestion
        SynchronizeVariable<array_1d<double, 9 >,double >(rThisVariable);
```

File: kratos/mpi/includes/mpi_communicator.h [hunk 10]
```diff
@@ -577,6 +595,24 @@ class MPICommunicator : public Communicator
         return true;
     }
 
+    bool SynchronizeNonHistoricalVariable(Variable<array_1d<double, 4 > > const& ThisVariable) override
```
Comment (adityaghantasala) [inline kratos/mpi/includes/mpi_communicator.h]:
```suggestion
    bool SynchronizeNonHistoricalVariable(Variable<array_1d<double, 4 > > const& rThisVariable) override
```

File: kratos/mpi/includes/mpi_communicator.h [hunk 11]
```diff
@@ -577,6 +595,24 @@ class MPICommunicator : public Communicator
...
+    {
+        SynchronizeNonHistoricalVariable<array_1d<double, 4 >,double>(ThisVariable);
```
Comment (adityaghantasala) [inline kratos/mpi/includes/mpi_communicator.h]:
```suggestion
        SynchronizeNonHistoricalVariable<array_1d<double, 4 >,double>(rThisVariable);
```

File: kratos/mpi/includes/mpi_communicator.h [hunk 12]
```diff
@@ -577,6 +595,24 @@ class MPICommunicator : public Communicator
...
+        return true;
+    }
+
+    bool SynchronizeNonHistoricalVariable(Variable<array_1d<double, 6 > > const& ThisVariable) override
```
Comment (adityaghantasala) [inline kratos/mpi/includes/mpi_communicator.h]:
```suggestion
    bool SynchronizeNonHistoricalVariable(Variable<array_1d<double, 6 > > const& rThisVariable) override
```

File: kratos/mpi/includes/mpi_communicator.h [hunk 13]
```diff
@@ -577,6 +595,24 @@ class MPICommunicator : public Communicator
...
+    {
+        SynchronizeNonHistoricalVariable<array_1d<double, 6 >,double>(ThisVariable);
```
Comment (adityaghantasala) [inline kratos/mpi/includes/mpi_communicator.h]:
```suggestion
        SynchronizeNonHistoricalVariable<array_1d<double, 6 >,double>(rThisVariable);
```

File: kratos/mpi/includes/mpi_communicator.h [hunk 14]
```diff
@@ -577,6 +595,24 @@ class MPICommunicator : public Communicator
...
+        return true;
+    }
+
+    bool SynchronizeNonHistoricalVariable(Variable<array_1d<double, 9 > > const& ThisVariable) override
```
Comment (adityaghantasala) [inline kratos/mpi/includes/mpi_communicator.h]:
```suggestion
    bool SynchronizeNonHistoricalVariable(Variable<array_1d<double, 9 > > const& rThisVariable) override
```

File: kratos/mpi/includes/mpi_communicator.h [hunk 15]
```diff
@@ -577,6 +595,24 @@ class MPICommunicator : public Communicator
...
+    {
+        SynchronizeNonHistoricalVariable<array_1d<double, 9 >,double>(ThisVariable);
```
Comment (adityaghantasala) [inline kratos/mpi/includes/mpi_communicator.h]:
```suggestion
        SynchronizeNonHistoricalVariable<array_1d<double, 9 >,double>(rThisVariable);
```

File: kratos/includes/communicator.h [hunk 16]
```diff
@@ -592,6 +592,21 @@ class Communicator
...
+    {
+        return true;
+    }
+
+    virtual bool SynchronizeVariable(Variable<array_1d<double, 6 > > const& ThisVariable)
+    {
+        return true;
+    }
+
+    virtual bool SynchronizeVariable(Variable<array_1d<double, 9 > > const& ThisVariable)
```
Comment (adityaghantasala) [inline kratos/includes/communicator.h]:
```suggestion
    virtual bool SynchronizeVariable(Variable<array_1d<double, 9 > > const& rThisVariable)
```

File: kratos/includes/communicator.h [hunk 17]
```diff
@@ -592,6 +592,21 @@ class Communicator
         return true;
     }
 
+    virtual bool SynchronizeVariable(Variable<array_1d<double, 4 > > const& ThisVariable)
+    {
+        return true;
+    }
+
+    virtual bool SynchronizeVariable(Variable<array_1d<double, 6 > > const& ThisVariable)
```
Comment (adityaghantasala) [inline kratos/includes/communicator.h]:
```suggestion
    virtual bool SynchronizeVariable(Variable<array_1d<double, 6 > > const& rThisVariable)
```

File: kratos/includes/communicator.h [hunk 18]
```diff
@@ -592,6 +592,21 @@ class Communicator
         return true;
     }
 
+    virtual bool SynchronizeVariable(Variable<array_1d<double, 4 > > const& ThisVariable)
```
Comment (adityaghantasala) [inline kratos/includes/communicator.h]:
```suggestion
    virtual bool SynchronizeVariable(Variable<array_1d<double, 4 > > const& rThisVariable)
```

**Review by adityaghantasala — CHANGES_REQUESTED** [review]

**Comment by philbucher** [comment]
done

**Review by adityaghantasala — DISMISSED** [review]

**Review by adityaghantasala — APPROVED** [review]

**Pull request merged**